src/Flexible.php

```diff
@@ -243,11 +241,15 @@ public function resolve($resource, $attribute = null)
     {
         $attribute = $attribute ?? $this->attribute;
 
-        $this->registerOriginModel($resource);
+        if (! $this->resolveCallback) {
+            $this->registerOriginModel($resource);
 
-        $this->buildGroups($resource, $attribute);
+            $this->buildGroups($resource, $attribute);
 
-        $this->value = $this->resolveGroups($this->groups);
+            $this->value = $this->resolveGroups($this->groups);
+        } elseif (is_callable($this->resolveCallback)) {
```

Comment:
This is purely code-style related and nothing to do with the functionality, but it would be nicer IMO to avoid the `elseif` and instead return early if the resolve callback is callable.

Something like 
```php
if ($this->resolveCallback && is_callable($this->resolveCallback)) {
    return $this->value = ...
}


```

Reply:
I'll change it to just a normal return void. The Field class documents it should return a void. Don't want to overwrite that and create any unnecessary side-effects.